src/rewriter.coffee

```diff
@@ -511,6 +511,9 @@ exports.Rewriter = class Rewriter
     @scanTokens (token, i, tokens) ->
       return 1 if     token[2]
       return 1 unless token.generated or token.explicit
+      if token.fromThen and token[0] is 'INDENT'
```

Comment:
The test case with an "empty `then` block" (`while 1 < 2 then`) called out the need to include this logic - without it, the while body block was being assigned empty location data before the `then`, whereas it should include the `then`